osu.Game.Tests/Visual/Gameplay/TestSceneSpectator.cs

```diff
@@ -214,6 +213,9 @@ public void TestHostFails()
             checkPaused(false); // Should continue playing until out of frames
             checkPaused(true); // And eventually stop after running out of frames and fail.
             // Todo: Should check for + display a failed message.
+
+            AddStep("exit player", () => InputManager.Key(Key.Escape));
+            AddAssert("player exited", () => Stack.CurrentScreen is SoloSpectatorScreen);
```

Comment:
Cherry-picking this to master with `SpectatorPlayer.AllowBackButton => false` does not fail.

Reply:
Resolved in